test/validation/api/ipsec/ipsec.c

```diff
@@ -815,10 +858,26 @@ static int ipsec_send_out_one(const ipsec_test_part *part,
 			ev = recv_event(queue, 0);
 			if (ODP_EVENT_INVALID != ev) {
 				CU_ASSERT(odp_event_is_valid(ev) == 1);
+
+				if (ODP_EVENT_IPSEC_STATUS == odp_event_type(ev) &&
+				    ODP_QUEUE_TYPE_SCHED == suite_context.q_type &&
+				    part->flags.test_sa_expiry) {
+					ipsec_status_event_handle(ev, sa, part->out[i].sa_expiry);
+					continue;
+				}
+
 				CU_ASSERT_EQUAL(ODP_EVENT_PACKET,
 						odp_event_types(ev, &subtype));
 				CU_ASSERT_EQUAL(ODP_EVENT_PACKET_BASIC,
 						subtype);
+
+				if (part->flags.test_sa_expiry) {
+					if (part->out[i].sa_expiry == IPSEC_TEST_EXPIRY_HARD_PKT ||
+					    part->out[i].sa_expiry == IPSEC_TEST_EXPIRY_HARD_BYTE) {
+						suite_context.status_notify = true;
+						return num_out;
```

Comment:
I do not understand this. Whenever hard lifetime expiry is being tested, any succesfully outbound inline IPsec processing causes the "status notify" flag to be set. So the flag would get set even if the expiration was never reported.

The return here means that some of the later asserts (about sa, sa, ctx, user_ptr) are skipped for no good reason.

The return also leaves the pkto output parameter unset, crashing the caller in some cases (e.g. if the ODP under test is not reporting hard lifetime expiry).